gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/SlsManifest.java

```diff
@@ -24,7 +24,7 @@
 
 @Value.Immutable
 @JsonSerialize(as = ImmutableSlsManifest.class)
-@JsonDeserialize(as = ImmutableSlsManifest.class)
+@JsonDeserialize(builder = SlsManifest.Builder.class)
```

Comment:
why was this changed?

Reply:
Because I centralized read/write calls into the Serializations class and added a reader metthod for SlsManifest.  Jackson throws exception without the builder - 
Caused by: java.lang.IllegalArgumentException: Cannot find a deserializer for non-concrete Map type [map type; class com.google.common.collect.ImmutableMap, [simple type, class java.lang.String] -> [simple type, class java.lang.Object]]
	at com.fasterxml.jackson.databind.deser.BasicDeserializerFactory.createMapDeserializer(BasicDeserializerFactory.java:1358)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer2(DeserializerCache.java:387)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer(DeserializerCache.java:349)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:264)
	... 13 more

It works in other test classes (see JavaServiceDistributionPluginTests) because they add a GuavaModule() to the object mapper configuration, but I didn't want to alter the production one.

Reply:
I think you should just add the guava module to the object mapper then :)

Reply:
not that I'm an expert on jackson deserialization, but if the tool provides a built in way to specify a builder, and the instances are supposed to be constructed with a builder, shouldn't we, uh, ya know, use a builder?

Reply:
I think its just a gotcha to remember specify the builder especially since it would fail at runtime. Its pretty standard for people to do `@JsonDeserialize(as = Immutable..)` and basically all of our object mappers are configured with the guava module. don't care too much but just seems easier to make the change once in the mapper